soroban-env-host/src/budget/model.rs

```diff
@@ -81,7 +81,7 @@ impl From<f64> for ScaledU64 {
     }
 }
 
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Copy, Debug, Default)]
```

Comment:
In general I do not want structs to be Copy. In this specific case it's not just a matter of having to call `clone()` somewhere, it's needed to allow the nontrivial array initializer, and the only workarounds for array-initializing a non-Copy array are all quite convoluted and awful.

Reply:
(See the change in BudgetDimension::Default that now manually initializes the `cost_models` field, because it's now longer than 32 elements and the `Derive(Default)` only works up to 32 elements. Silly causal chain here.)